server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregator.java

```diff
@@ -287,7 +287,7 @@ private boolean subAggsNeedScore() {
     }
 
     @Override
-    protected boolean shouldDefer(Aggregator aggregator) {
+    public boolean shouldDefer(Aggregator aggregator) {
```

Comment:
Is there another way to test this function without changing the visibility ? I understand it's for unit tests but the first condition is always true for significant terms and we don't verify the other condition in the added uts so is there a value to do this ?

Reply:
I think I'll just remove the test and avoid making this public -- the test was more valuable in previous versions of the PR when we were dynamically selecting the collection mode.

Reply:
+1, thanks